src/System.Private.CoreLib/src/System/Runtime/CompilerServices/ContractHelper.CoreCLR.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Contracts;
+
+namespace System.Runtime.CompilerServices
+{
+    public static partial class ContractHelper
+    {
+        /// <summary>
+        /// Rewriter calls this method to get the default failure behavior.
+        /// </summary>
+        [System.Diagnostics.DebuggerNonUserCode]
+        public static void TriggerFailure(ContractFailureKind kind, string displayMessage, string userMessage, string conditionText, Exception innerException)
```

Comment:
It should be fine to just call `Debug.ContractFailure` in both CoreCLR and CoreRT, and not have this runtime-specific file.

Reply:
`Debug.ContractFailure` doesn't exist in CoreRT today. I thought about adding it with the old `TriggerFailure` code as implementation, but it doesn't pass all the required parameters. Eventually the CoreRT implementation calls some external code that seems to be a pluggable debugging interface. I was not comfortable with breaking it.

Reply:
The CoreRT implementation: https://github.com/dotnet/corert/pull/6838/files#diff-405ecc452c685c4dbb5735598afaa4d2